### DISCUSSION TITLE
Dynamically added images not showing up

### DISCUSSION THREAD

**Discussion by ViktorVojtek** [vercel/next.js#12554] [Help]
In next >= 9.3 the dynamically added images not showing up after project build.

I have custom nodejs server where I create products and uploading product images to public folder.
These product images are not visible on pages till I manually rebuild the whole project with 'next build'.

when I open the image path in the browser the next return 404 page.

In the previous versions of next everything works fine.

## Screenshots

<img width="550" alt="Screenshot 2020-05-06 at 12 47 25" src="https://user-images.githubusercontent.com/13499872/81168780-f31fc280-8f97-11ea-8479-0e93b04f22fc.png">

<img width="980" alt="Screenshot 2020-05-06 at 12 49 22" src="https://user-images.githubusercontent.com/13499872/81168832-0a5eb000-8f98-11ea-9509-71da70d5d0c9.png">

<img width="949" alt="Screenshot 2020-05-06 at 12 49 58" src="https://user-images.githubusercontent.com/13499872/81168877-1d718000-8f98-11ea-9d23-95587b5ac8a7.png">

## System information

- OS: [e.g. macOS, ubuntu]
- Browser (if applies) [e.g. chrome]
- Version of Next.js: [e.g. 9.3.1]
- Version of Node.js: [e.g. 10.3.0]

**Comment by alisyahidin** [comment]
Got the same issue here. Still finding the answer and hopefully someone can help :)

**Reply by ViktorVojtek** [reply]
Hey Ali, creating the old "static" folder for images would fix that problem. The public folder just not working as expected for now.

**Reply by MacCode7110** [reply]
I am currently having this issue; do you mean renaming the public folder to "static"

**Reply by ViktorVojtek** [reply]
Hey @MacCode7110 yes, exactly.

**Reply by sprudzans** [reply]
> Hey @MacCode7110 yes, exactly.

No, that's not a solution. It's also like public folder in dev mode , but in build not working

**Comment by kasipavankumar** [comment]
**Files added to the `public` directory during the runtime won't be available in the application.** A cloud storage service like S3 or Google Cloud Storage can be used to store user generated or runtime uploaded images. 

Also, since you are using a custom server, upload images to any directory other than `public` & use the [Multer](https://www.npmjs.com/package/multer) middleware for handling those uploads.

[Documentation](https://nextjs.org/docs/basic-features/static-file-serving#:~:text=Only%20assets%20that%20are%20in%20the%20public%20directory%20at%20build%20time%20will%20be%20served%20by%20Next.js.) (Note)

**Reply by wlisesrivas** [reply]
So why it works locally (on dev) but not on prod, they should persist the same behavior to avoid developer says "it works on my machine"
